src/AutoRest.CSharp/Common/Output/Models/Types/BuildContextOfT.cs

```diff
@@ -24,11 +22,12 @@ private T EnsureLibrary()
             T library;
             if (Configuration.Generation1ConvenienceClient)
             {
-                library = (T)(object)new DataPlaneOutputLibrary(CodeModel, (BuildContext<DataPlaneOutputLibrary>)(object)this);
+                throw new InvalidOperationException($"{nameof(BuildContext)} isn't supported in legacy DataPlane");
             }
-            else if (Configuration.AzureArm)
+
+            if (Configuration.AzureArm)
             {
-                library = (T)(object)new MgmtOutputLibrary();
+                library = (T)(object)new MgmtOutputLibrary(CodeModel, SchemaUsageProvider);
```

Comment:
I believe once we could support typespec in mgmt we could make mgmt no longer supports this BuildContext as well and we could remove this type then, is it?

Reply:
Yes